example/src/new_api/swipeable/Swipeable.tsx

```diff
@@ -0,0 +1,558 @@
+        rowState.value === 1
+          ? leftWidth.value
+          : rowState.value === -1
+          ? -rightWidth.value
+          : 0;
+
+      const offsetDrag = userDrag.value / friction + startOffset;
+
+      appliedTranslation.value = interpolate(
+        offsetDrag,
+        [
+          -rightWidth.value - 1,
+          -rightWidth.value,
+          leftWidth.value,
+          leftWidth.value + 1,
+        ],
+        [
+          -rightWidth.value - (overshootRight ? 1 / overshootFriction : 0),
+          -rightWidth.value,
+          leftWidth.value,
+          leftWidth.value + (overshootLeft ? 1 / overshootFriction : 0),
+        ]
+      );
+
+      showLeftProgress.value =
+        leftWidth.value > 0
+          ? interpolate(
+              appliedTranslation.value,
+              [-1, 0, leftWidth.value],
+              [0, 0, 1]
+            )
+          : 0;
+      leftActionTranslate.value = interpolate(
+        showLeftProgress.value,
+        [0, Number.MIN_VALUE],
+        [-10000, 0],
+        Extrapolation.CLAMP
+      );
+      showRightProgress.value =
+        rightWidth.value > 0
+          ? interpolate(
+              appliedTranslation.value,
+              [-rightWidth.value, 0, 1],
+              [1, 0, 0]
+            )
+          : 0;
+      rightActionTranslate.value = interpolate(
+        showRightProgress.value,
+        [0, Number.MIN_VALUE],
+        [-10000, 0],
+        Extrapolation.CLAMP
+      );
+    };
+
+    const animateRow = (
+      fromValue: number,
+      toValue: number,
+      velocityX?: number
+    ) => {
+      'worklet';
+
+      rowState.value = Math.sign(toValue);
+
+      appliedTranslation.value = withSpring(
+        toValue,
+        {
+          duration: 1000,
+          dampingRatio: 1.2,
+          stiffness: 500,
+          velocity: velocityX,
+          ...props.animationOptions,
+        },
+        (isFinished) => {
+          if (isFinished) {
+            if (toValue > 0 && props.onSwipeableOpen) {
+              runOnJS(props.onSwipeableOpen)('left', swipeableMethods);
+            } else if (toValue < 0 && props.onSwipeableOpen) {
+              runOnJS(props.onSwipeableOpen)('right', swipeableMethods);
+            } else if (props.onSwipeableClose) {
+              const closingDirection = fromValue > 0 ? 'left' : 'right';
+              runOnJS(props.onSwipeableClose)(
+                closingDirection,
+                swipeableMethods
+              );
+            }
+          }
+        }
+      );
+
+      if (toValue > 0 && props.onSwipeableWillOpen) {
+        runOnJS(props.onSwipeableWillOpen)('left');
+      } else if (toValue < 0 && props.onSwipeableWillOpen) {
+        runOnJS(props.onSwipeableWillOpen)('right');
+      } else if (props.onSwipeableWillClose) {
+        const closingDirection = fromValue > 0 ? 'left' : 'right';
+        runOnJS(props.onSwipeableWillClose)(closingDirection);
+      }
+    };
+
+    const handleRelease = (
```

Comment:
This is used only in one place and can be moved much closer to where it's called (or even be inlined)